iree/hal/vulkan/vulkan_device.cc

```diff
@@ -910,6 +910,17 @@ static iree_hal_allocator_t* iree_hal_vulkan_device_allocator(
   return device->device_allocator;
 }
 
+static iree_status_t iree_hal_vulkan_device_query_i32(
+    iree_hal_device_t* base_device, iree_string_view_t key,
+    int32_t* out_value) {
+  // iree_hal_vulkan_device_t* device =
+  //     iree_hal_vulkan_device_cast(base_device);
+  *out_value = 0;
+  return iree_make_status(IREE_STATUS_NOT_FOUND,
+                          "unknown device configuration key value '%*.s'",
+                          (int)key.size, key.data);
```

Comment:
I could see this also returning `IREE_STATUS_UNIMPLEMENTED`, though always returning "key not found" is technically correct when no keys are handled :P

Reply:
yep - NOT_FOUND is the only observable behavior here - this could even be a bool return but having the status means we get a nice readable error message :)